pkg/peer.go

```diff
@@ -0,0 +1,154 @@
+package sfu
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/pion/ion-sfu/pkg/log"
+	"github.com/pion/webrtc/v3"
+)
+
+var (
+	// ErrTransportExists join is called after a peerconnection is established
+	ErrTransportExists = errors.New("rtc transport already exists for this connection")
+	// ErrNoTransportEstablished cannot signal before join
+	ErrNoTransportEstablished = errors.New("no rtc transport exists for this Peer")
+)
+
+// Peer represents a single peer signal session
+type Peer struct {
+	sfu *SFU
+	pc  *WebRTCTransport
+
+	Done      chan bool
+	IceChan   chan *webrtc.ICECandidate
+	OfferChan chan *webrtc.SessionDescription
+}
+
+// NewPeer creates a new Peer for signaling with the given SFU
+func NewPeer(sfu *SFU) Peer {
+	return Peer{
+		sfu:       sfu,
+		Done:      make(chan bool),
+		IceChan:   make(chan *webrtc.ICECandidate),
+		OfferChan: make(chan *webrtc.SessionDescription),
```

Comment:
I would prefer to use plain callbacks here. Since we don't need to worry about concurrency so chans add a bunch of overhead. Otherwise looking good!